src/app/loadout/plug-drawer/PlugDrawer.tsx

```diff
@@ -243,25 +252,35 @@ export default function PlugDrawer({
   );
 }
 
+/**
+ * A map of plugSetHashes to the selected plugs for that plugSetHash.
+ */
+type InternalSelectedState = {
+  [plugSetHash: number]: PluggableInventoryItemDefinition[] | undefined;
+};
```

Comment:
I was thinking about it and maybe we are better off expecting this data structure as our`initiallySelected` prop. 

The issue as I see it is we have a flat list of mods in the API and now we need those mods indexed by a `plugSetHash` and we just need to find the most appropriate point to do that. My initial cut was to hide it away from implementers of this `PlugDrawer` component so it is only done once. But maybe it makes more sense for the consumer of the component to decide how that is done. 

Thoughts?